api/app/Models/User.php

```diff
@@ -930,6 +943,35 @@ public static function scopeGeneralSearch(Builder $query, ?array $searchTerms):
                 // Now that we have added a column, query builder no longer will add a * to the select.  Add all possible columns manually.
                 ->addSelect(self::$selectableColumns);
 
+            // negation setup
+            preg_match('/\-[a-zA-Z]+\b/', $combinedSearchTerm, $negationMatches);
```

Comment:
This regex can be improved. First of all, `!` can also indicate negation. Second, we don't want to ignore all non-letter characters in search terms (eg hyphens or apostrophes in names, or `@`, underscore, or dot in emails).

Try these inputs in your favourite regex tester (I use https://www.regexplanet.com/advanced/javascript/index.html)
```
john -smith
john !smith
john -bob@test.com
blast-from-the-past
-john smith
```
I think this regex will work: `(^|\s)[-!][^\s]+\b`

Reply:
Regex, what a wonderful world...
I'll roll with your suggestion, it seems perfect
https://github.com/GCTC-NTGC/gc-digital-talent/pull/9977/commits/cad0c49378aabeee5baefd6478f2239eac4f7640